static/templates/login_to_access.hbs

```diff
@@ -0,0 +1,22 @@
+<div id="login_to_access_modal" class="modal modal-bg fade new-style" tabindex="-1" role="dialog" aria-labelledby="login_to_access_modal" aria-hidden="true">
+    <div class="modal-header">
+
+        <div class="color_animated_button go_back_button">
+            <i class="fa fa-arrow-left"></i>
+            <span class="login_to_access_label"><b>{{t "Access restricted" }}</b></span>
+        </div>
+    </div>
+    <div class="modal-body">
+        <p>{{t "You are not signed in right now. Hence, you only have read access to messages in" }}
+            <a href="https://zulipchat.com/help/stream-permissions">{{t "Web public streams" }}</a>
+            {{t "and the ability to search among those messages." }}
+        </p>
+        <div class="topic_edit_spinner"></div>
+    </div>
```

Comment:
Can you add a screenshot of this modal, possibly in a chat.zulip.org thread?  I suspect we'll be OK with merging this even if it's imperfect UI since it's a good place to iterate from, but it'd be nice to at least have some discussion of what strings we want.

@alya FYI.

Reply:
Started discussion in https://chat.zulip.org/#narrow/stream/101-design/topic/web.20public.20view.20login.20required.20modal.20.2318532